src/linear_algebra/matrix/nonsingular_inverse.lean

```diff
@@ -36,11 +57,111 @@ open equiv equiv.perm finset
 
 variables (A : matrix n n α) (B : matrix n n α)
 
+/-! ### Matrices are `invertible` iff their determinants are -/
+
+section invertible
+
+/-- A copy of `inv_of_mul_self` using `⬝` not `*`. -/
+protected lemma inv_of_mul_self [invertible A] : ⅟A ⬝ A = 1 := inv_of_mul_self A
+
+/-- A copy of `mul_inv_of_self` using `⬝` not `*`. -/
+protected lemma mul_inv_of_self [invertible A] : A ⬝ ⅟A = 1 := mul_inv_of_self A
+
+/-- If `A.det` has a constructive inverse, produce one for `A`. -/
+def invertible_of_det_invertible [invertible A.det] : invertible A :=
+{ inv_of := ⅟A.det • A.adjugate,
+  mul_inv_of_self :=
+    by rw [mul_smul_comm, matrix.mul_eq_mul, mul_adjugate, smul_smul, inv_of_mul_self, one_smul],
+  inv_of_mul_self :=
+    by rw [smul_mul_assoc, matrix.mul_eq_mul, adjugate_mul, smul_smul, inv_of_mul_self, one_smul] }
+
+lemma inv_of_eq [invertible A.det] [invertible A] : ⅟A = ⅟A.det • A.adjugate :=
+by { letI := invertible_of_det_invertible A, convert (rfl : ⅟A = _) }
+
+/-- `A.det` is invertible if `A` has a left inverse. -/
+def det_invertible_of_left_inverse (h : B ⬝ A = 1) : invertible A.det :=
+{ inv_of := B.det,
+  mul_inv_of_self := by rw [mul_comm, ← det_mul, h, det_one],
+  inv_of_mul_self := by rw [← det_mul, h, det_one] }
+
+/-- `A.det` is invertible if `A` has a right inverse. -/
+def det_invertible_of_right_inverse (h : A ⬝ B = 1) : invertible A.det :=
+{ inv_of := B.det,
+  mul_inv_of_self := by rw [← det_mul, h, det_one],
+  inv_of_mul_self := by rw [mul_comm, ← det_mul, h, det_one] }
+
+/-- If `A` has a constructive inverse, produce one for `A.det`. -/
+def det_invertible_of_invertible [invertible A] : invertible A.det :=
```

Comment:
Can this be an instance?
And do we want an `iff` version of this statement?

Reply:
This was just moved from further down the file. I don't think we make a habit of actually making global `invertible` instances.

It's data so can't be an `iff`. I plan to make a follow-up PR with the equiv